src/components/SlateEditor/plugins/related/RelatedArticleBox.tsx

```diff
@@ -6,235 +6,203 @@
 import { Editor, Transforms } from 'slate';
 import { ReactEditor, RenderElementProps } from 'slate-react';
 import { useTranslation } from 'react-i18next';
 import styled from '@emotion/styled';
-import compact from 'lodash/compact';
-import { RelatedArticleList } from '@ndla/ui';
-import { toggleRelatedArticles } from '@ndla/article-scripts';
-import { IArticle } from '@ndla/types-backend/draft-api';
-import { convertFieldWithFallback } from '../../../../util/convertFieldWithFallback';
+import { RelatedContentEmbedData, RelatedContentMetaData } from '@ndla/types-embed';
+import { RelatedArticleListV2, RelatedContentEmbed } from '@ndla/ui';
+import { IconButtonV2 } from '@ndla/button';
+import { Pencil, TrashCanOutline } from '@ndla/icons/action';
 import { fetchDraft } from '../../../../modules/draft/draftApi';
-import { queryResources } from '../../../../modules/taxonomy';
 import EditRelated from './EditRelated';
-import handleError from '../../../../util/handleError';
-import RelatedArticle from './RelatedArticle';
 import { RelatedElement } from '.';
-import { Resource } from '../../../../modules/taxonomy/taxonomyApiInterfaces';
-import { ARTICLE_EXTERNAL } from '../../../../constants';
 import { useTaxonomyVersion } from '../../../../containers/StructureVersion/TaxonomyVersionProvider';
+import { fetchNodes } from '../../../../modules/nodes/nodeApi';
 
 interface Props {
   attributes: RenderElementProps['attributes'];
   editor: Editor;
   element: RelatedElement;
-  locale?: string;
   onRemoveClick: (e: MouseEvent<HTMLButtonElement>) => void;
   children: ReactNode;
 }
 
-const StyledDiv = styled.div`
-  & article > p {
-    font-family: Source Sans Pro !important;
-  }
-`;
-
-export interface ExternalArticle {
-  id: 'external-learning-resources';
-  tempId: string;
-  url: string;
-  title: string;
-  description: string;
-}
+const externalEmbedToMeta = async (
+  embedData: RelatedContentEmbedData,
+  seq: number,
+): Promise<RelatedContentMetaData> => {
+  return {
+    resource: 'related-content',
+    embedData,
+    seq,
+    status: 'success',
+    data: undefined,
+  };
+};
 
-interface InternalArticle extends Omit<IArticle, 'title' | 'id'> {
-  resource: Resource[];
-  id: string;
-  title: string;
-}
+const internalEmbedToMeta = async (
+  embedData: RelatedContentEmbedData,
+  seq: number,
+  language: string,
+  taxonomyVersion: string,
+): Promise<RelatedContentMetaData> => {
+  const article = await fetchDraft(embedData.articleId!, language).catch(() => undefined);
+  const nodes = await fetchNodes({
+    taxonomyVersion,
+    contentURI: `urn:article:${embedData.articleId}`,
+    language: language,
+  }).catch(() => undefined);
+
+  if (!!article && !!nodes?.length) {
+    return {
+      resource: 'related-content',
+      seq,
+      embedData,
+      status: 'success',
+      data: {
+        article,
+        resource: nodes[0]!,
+      },
+    };
+  } else {
+    return {
+      resource: 'related-content',
```

Comment:
'related-content' kunne vore en const i og med at den brukes mange plasser i fila. 'success' også forsåvidt.

Reply:
Har egentlig unngått det med vilje. Med typesikkerhet trenger man nesten aldri å skrive hele verdien.

Grunnen til at jeg har unngått det er at slate bruker en hel haug med constants for å beskrive typene til en node. Disse har strengt tatt ingenting å gjøre med embed-dataen, og er ofte ikke helt like. Her er f.eks `TYPE_RELATED = 'related'`, mens `resource`-typen til `RelatedContent` er `related-content`. Tenkte det kom til å unngå litt forvirring. Hva tenker du?

Reply:
Så lenge du har en tanke bak kvifor du har gjort det slik er eg fornøgd.